### PR TITLE
Initial support for image scanning

### LINKED ISSUE #927: Add Support For ECR

### PR DESCRIPTION
**Pull request by Rchanger**
Initial support for image scanning. closes #927

### CODE REVIEW

File: pkg/iac-providers/output/vulnerability.go [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+// ACRVendorInfo vendor details
+type ACRVendorInfo struct {
+	Title string `json:"title"`
+	Link  string `json:"link"`
+}
+
+// ACRCvss cvss score
+type ACRCvss struct {
+	Two0   ACRCvssBase `json:"2.0"`
```
Comment (patilpankaj212) [inline pkg/iac-providers/output/vulnerability.go:123]: can these fields changed to `V2` and `V3`?

File: pkg/iac-providers/output/vulnerability.go [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+	Three0 ACRCvssBase `json:"3.0"`
+}
+
+//ACRCvssBase ...
```
Comment (patilpankaj212) [inline pkg/iac-providers/output/vulnerability.go:127]: please add a description for the struct.

File: pkg/runtime/executor.go [hunk 3]
```diff
@@ -307,6 +323,26 @@ func (e *Executor) findViolations(results *Output) error {
 	return nil
 }
 
+// getVulnerabilities add vulnerability findings to output
+func (e *Executor) getVulnerabilities(results *Output, options map[string]interface{}) output.AllResourceConfigs {
+
+	res := e.vulnerabilityEngine.FindVulnerabilities(results.ResourceConfig, options)
+
+	return res
```
Comment (patilpankaj212) [inline pkg/runtime/executor.go:331]: can be modified to `return e.vulnerabilityEngine.FindVulnerabilities(results.ResourceConfig, options)`

File: pkg/iac-providers/output/types.go [hunk 4]
```diff
@@ -44,8 +44,9 @@ type ResourceConfig struct {
 
 // ContainerNameAndImage container name and image
 type ContainerNameAndImage struct {
-	Name  string `json:"name"`
-	Image string `json:"image"`
+	Name            string          `json:"name"`
+	Image           string          `json:"image"`
+	Vulnerabilities []Vulnerability `json:"vulnerabilities"`
```
Comment (patilpankaj212) [inline pkg/iac-providers/output/types.go:49]: should the struct's name be changed as it now contain additional field? eg: `ContainerDetails` ?

**Review by patilpankaj212 — DISMISSED** [review]
LGTM !!

**Review by cesar-rodriguez — CHANGES_REQUESTED** [review]
Can you please fix merge conflicts?

**Review by cesar-rodriguez — CHANGES_REQUESTED** [review]
Can you please update the readme and usage docs with instructions for the new `--find-vuln` flag?

**Comment by Rchanger** [comment]
> Can you please update the readme and usage docs with instructions for the new `--find-vuln` flag?

updated  readme and usage docs

**Review by cesar-rodriguez — APPROVED** [review]

**Pull request merged**